pandas/tests/frame/test_timeseries.py

```diff
@@ -323,6 +323,14 @@ def test_asfreq_datetimeindex(self):
         ts = df['A'].asfreq('B')
         tm.assertIsInstance(ts.index, DatetimeIndex)
 
+    def test_asfreq_datetimeindex_empty_series(self):
+        # GH 14340
+        expected = Series(index=pd.DatetimeIndex(
+            ["2016-09-29 11:00"])).asfreq('H')
+        result = Series(index=pd.DatetimeIndex(["2016-09-29 11:00"]),
+                        data=[3]).asfreq('H')
+        self.assert_index_equal(expected.index, result.index)
+
```

Comment:
compare the series, e.g. ``assert_series_equal``, and construct the expected directly.

you have two results to compare here, what you currently have labeled as expected and as result.

make another tests where you construct the expected directly (iow, dont' use ``.asfreq``) . IOW you actually should use a direc